api/lambda/database/models/api.go

```diff
@@ -0,0 +1,156 @@
+	GetTablesDetail      *GetTablesDetailInput      `json:"getTablesDetail"`
+}
+
+type GetDatabasesInput struct {
+	DatabaseName *string `json:"databaseName,omitempty"` // if empty get all databases
+}
+
+// NOTE: we will assume this is small an not paginate
+type GetDatabasesOutput struct {
+	Error
+	GetDatabasesInput
+	Databases []*DatabaseDescription `json:"databases,omitempty"`
+}
+
+type DatabaseDescription struct {
+	DatabaseName string  `json:"databaseName"`
+	Description  *string `json:"description,omitempty"`
+	// other stuff? CreateDate?
+}
+
+type GetTablesInput struct {
+	DatabaseName string `json:"databaseName" validate:"required"`
+	HavingData   bool   `json:"havingData,omitempty"` // if true, only return table containing data
+}
+
+// NOTE: we will assume this is small an not paginate
+type GetTablesOutput struct {
+	Error
+	GetTablesInput
+	Tables []*TableDescription `json:"tables,omitempty"`
+}
+
+type TableDescription struct {
+	DatabaseName string  `json:"databaseName"`
+	TableName    string  `json:"tableName"`
+	Description  *string `json:"description,omitempty"`
+	// other stuff? CreateDate?
+}
+
+type TableDetail struct {
+	TableDescription
+	Columns []*TableColumn `json:"columns,omitempty"`
+}
+
+type GetTablesDetailInput struct {
+	DatabaseName string   `json:"databaseName" validate:"required"`
+	TableNames   []string `json:"tableNames" validate:"required"`
+}
+
+// NOTE: we will assume this is small an not paginate
+type GetTablesDetailOutput struct {
+	Error
+	TablesDetails []*TableDetail `json:"tables_details,omitempty"`
+}
+
+type TableColumn struct {
+	Name        string  `json:"name,omitempty" validate:"required"`
+	Type        string  `json:"type,omitempty" validate:"required"`
+	Description *string `json:"description,omitempty"`
+}
+
+type ExecuteAsyncQueryInput struct {
+	DatabaseName       string `json:"databaseName" validate:"required"`
+	SQL                string `json:"sql" validate:"required"`
+	ResultsMaxPageSize *int64 `json:"resultsMaxPageSize"` // only return this many per call
+}
+
+type ExecuteAsyncQueryOutput struct {
+	Error
+	GetQueryResultsOutput // might be filled in if query ran fast
+}
+
+type GetQueryStatusInput struct {
+	QueryID string `json:"queryId" validate:"required"`
+}
+
+type GetQueryStatusOutput struct {
+	Error
+	GetQueryStatusInput
+	Status string `json:"status" validate:"required,oneof=running,succeeded,failed"`
+}
+
+type GetQueryResultsInput struct {
+	QueryID            string  `json:"queryId" validate:"required"`
+	PaginationToken    *string `json:"paginationToken,omitempty"`
+	ResultsMaxPageSize *int64  `json:"resultsMaxPageSize"` // only return this many per call
+}
+
+type GetQueryResultsOutput struct {
+	Error
+	GetQueryResultsInput
+	Status  string `json:"status" validate:"required,oneof=running,succeeded,failed"`
+	NumRows int    `json:"numRows"`
+	Rows    []*Row `json:"rows"`
+}
+
+// Blocking query
+type ExecuteQueryInput ExecuteAsyncQueryInput
```

Comment:
I prefer making these type aliases so you can freely exchange them - makes it easier to share helper methods and such (here and elsewhere in this file)

```suggestion
type ExecuteQueryInput = ExecuteAsyncQueryInput
```

Reply:
Done